sirepo/job_driver/sbatch.py

```diff
@@ -87,33 +87,32 @@ def init_class(cls, job_supervisor_module):
 
     async def prepare_send(self, op):
         m = op.msg
-        try:
-            self._creds = m.pkdel('sbatchCredentials')
-            if self._srdb_root is None:
-                if not self._creds or 'username' not in self._creds:
-                    self._raise_sbatch_login_srexception('no-creds', m)
-                self._srdb_root = self.cfg.srdb_root.format(
-                    sbatch_user=self._creds.username,
-                )
-            m.userDir = '/'.join(
-                (
-                    str(self._srdb_root),
-                    sirepo.simulation_db.USER_ROOT_DIR,
-                    m.uid,
-                )
+        c = m.pkdel('sbatchCredentials')
+        if self._srdb_root is None or c:
+            if c:
+                self._creds = c
```

Comment:
If someone were to hack our api and send a message with sbatchCredentials on it when an agent is already started then they will never be deleted from the driver (self._creds). We don't log self._creds and we deleted them from the message so I don't think it is a big deal but worth mentioning.

Reply:
Correct.